cmd/image-builder/main.go

```diff
@@ -97,6 +99,37 @@ func ostreeImageOptions(cmd *cobra.Command) (*ostree.ImageOptions, error) {
 	}, nil
 }
 
+type registrations struct {
+	Redhat struct {
+		Subscription *subscription.ImageOptions `json:"subscription,omitempty"`
+	} `json:"redhat,omitempty"`
+}
+
+func subscriptionImageOptions(cmd *cobra.Command) (*subscription.ImageOptions, error) {
+	regFilePath, err := cmd.Flags().GetString("registrations")
+	if err != nil {
+		return nil, err
+	}
+	if regFilePath == "" {
+		return nil, nil
+	}
+
+	f, err := os.Open(regFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open registrations file: %w", err)
+	}
+	defer f.Close()
+
+	// XXX: support yaml eventually
```

Comment:
Why, why it is JSON first YAML maybe and not the other way around (on a new project)? :-)

Reply:
Mostly consistency with existing tech, current blueprints are json/toml so json seems okay, but given that blueprints++ will be YAML I'm fine moving to yaml here already.